lib/mdl/doc.rb

```diff
@@ -43,13 +43,17 @@ def initialize(text, ignore_front_matter = false)
     end
 
     ##
-    # Alternate 'constructor' passing in a filename
+    # Alternate 'constructor' passing in a filename. Exists with 1 if file
+    # doesn't exist
 
     def self.new_from_file(filename, ignore_front_matter = false)
       if filename == '-'
         new($stdin.read, ignore_front_matter)
-      else
+      elsif File.exist?(filename)
         new(File.read(filename, :encoding => 'UTF-8'), ignore_front_matter)
+      else
+        $stderr << "#{Errno::ENOENT}: No such file or directory - #{filename}"
+        exit 1
```

Comment:
the doc library shouldn't be exiting. the CLI should

Reply:
That would require another file read, is that ok?

Reply:
You don't need to read the file, just stat it (`File.exist?(...)`). Definitely don't read it twice... but you're already doing a stat here, moving it up a level isn't any less efficient.

Reply:
Ah I see. I'll work on that right away. Thanks for your help!